src/czbenchmarks/cli/cli_run.py

```diff
@@ -56,81 +49,196 @@ def run(ctx):
     """Run a benchmark task."""
     if ctx.invoked_subcommand is None:
         click.echo("Please specify a task to run, e.g., 'czbenchmarks run clustering'.")
-        click.echo(f"Available tasks: {', '.join(TASK_REGISTRY.keys())}")
+        click.echo(f"Available tasks: {', '.join(TASK_REGISTRY.list_tasks())}")
+
 
+# --- Dynamic Command Creation ---
 
-for task_name, task_def in TASK_REGISTRY.items():
-    # Discover parameters at definition time
-    task_params = discover_task_parameters(task_def)
-    baseline_params = discover_baseline_parameters(task_def)
+for task_name in TASK_REGISTRY.list_tasks():
+    task_info = TASK_REGISTRY.get_task_info(task_name)
 
-    @click.command(name=task_name, help=task_def.description)
-    @add_options(task_params)
-    @add_options(baseline_params)
-    @common_task_options
+    @click.command(name=task_name, help=task_info.description)
     def task_callback(**kwargs):
```

Comment:
This may also be a follow-up ticket, but `click` does a lot of the validation typecasting (including paths) for you if the command arguments have type annotations. Using `**kwargs` means you have to have the extra code to do it here. You have enough information from the task params to construct the actual signature for this, but it's a little tricky in terms of metaprogramming, so maybe for later.

Reply:
Oh, I see you're adding them with decorators below. Hmmm. Maybe that's fine and we just need to actually specify the types to click there.